server/request_handler.go

```diff
@@ -137,3 +144,10 @@ func (r *RpcRequestHandler) processBatchRequest(client RPCProxyClient, jsonBatch
 	// Write consolidated response
 	r._writeRpcBatchResponse(response)
 }
+
+func (r *RpcRequestHandler) finishRequest() {
+	timeRequestNeeded := time.Since(r.timeStarted) // At end of request, log the time it needed
+	r.requestEntry.RequestDuration = timeRequestNeeded
+	r.db.SaveRequestEntry(r.requestEntry)
```

Comment:
the database should be able to take the whole record and save the request and each individual ethsendrawtxreq from there

Reply:
I also thought about it in the beginning and This will not affect single request.But this logic will have extra looping & extra list/channel for batch request.More changes.

Pros
1.db is not passed everywhere.
2.Save db is called from a single place.

Cons
1.In the Current implementation each individual eth_sendRawRequests are saved inside the request itself and not much change required except the db is constructed during the request init phase. If we want to do it in the handler then we will have to return each eth_sendRawRequest and consolidate inside the list.Similar to r.JsonRes. Also extra list and channel needed.Method signature needs to be changed for  
`(r *RpcRequest) ProcessRequest()`
  method . 
2. During the finishRequest phase,the list needs to be passed inside the db method(SaveEthSendRawTxEntry) where we will have to loop in to store each individual eth_sendRawRequest.

Reply:
i think the EthSendRawTxEntry should probably be created by the handler and the pointer passed to the request. the RequestRecord should contain everything, and the database should be able to save everything within one batched request.